Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,356 @@
+            break;        
+        }
+
+        # Fetching cloud service details for remediation.
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.FailedControlSet
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId};
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($resourceDetails.ResourceDetails | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No cloud service(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        $resourceDetails.ResourceDetails | ForEach-Object { 
+            try
+            {
+                $resources += Get-AzResource -ResourceId $_.ResourceId
+            }
+            catch
+            {
+                Write-Host "Valid resource group(s) or resource name(s) not found in input json file. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                break
+            }
+        }
+    }
+
+    $totalCloudService = ($resources | Measure-Object).Count
+    if($totalCloudService -eq 0)
+    {
+        Write-Host "No cloud service(s) found." -ForegroundColor $([Constants]::MessageType.Error);
+        Write-Host $([Constants]::DoubleDashLine)
+        break;
+    }
+
+    Write-Host "Total cloud service(s): [$($totalCloudService)]"
+    $folderPath = [Environment]::GetFolderPath("LocalApplicationData") 
+    if (Test-Path -Path $folderPath)
+    {
+        $folderPath += "\AzTS\Remediation\Subscriptions\$($subscriptionid.replace("-","_"))\$((Get-Date).ToString('yyyyMMdd_hhmm'))\DisableRemoteDesktopAccess"
+        New-Item -ItemType Directory -Path $folderPath | Out-Null
+    }
+
+    Write-Host "Checking config of cloud service(s) for remediation: [$(($resources | Measure-Object).Count)]"
+    
+    try {
+        $cloudServiceWithEnabledRDPAccess = @()
+        $cloudServiceWithDisabledRDPAccess = @()
+        $skippedCloudServiceFromRemediation = @()
+
+        # Checking cloud service(s) with enabled RDP acccess
+        $resources |  ForEach-Object {
+            $resource = $_
+            $rdpExtensions = @()
+            
+            # Fetching cloud service(s) slot i.e. Staging, Production 
+            $cloudServiceSlots = Get-AzResource -ResourceGroupName $resource.ResourceGroupName -Name $resource.Name -ResourceType "$resourceType/slots" -ApiVersion "2016-04-01"
+            $cloudServiceSlotsName = $cloudServiceSlots.Name | sort -Unique
+            
+            # Checking RDP extension on each slots
+            $cloudServiceSlotsName | ForEach-Object {
+                $slot = $_
+                $cloudServiceExtension = Get-AzureServiceRemoteDesktopExtension -ServiceName $resource.Name -Slot $slot -ErrorAction SilentlyContinue
```

Comment:
Does this return RDP extension for all roles in a slot?

Reply:
Yes